test/redis/cache_impl_test.go

```diff
@@ -529,3 +420,107 @@ func TestRedisWithJitter(t *testing.T) {
 	assert.Equal(uint64(0), limits[0].Stats.OverLimit.Value())
 	assert.Equal(uint64(0), limits[0].Stats.NearLimit.Value())
 }
+
+func mustNewRedisServer() *miniredis.Miniredis {
+	srv, err := miniredis.Run()
+	if err != nil {
+		panic(err)
+	}
+
+	return srv
+}
+
+func TestNewClientImpl(t *testing.T) {
+	redisAuth := "123"
+	statsStore := stats.NewStore(stats.NewNullSink(), false)
+
+	mkRedisClient := func(auth, addr string) redis.Client {
+		return redis.NewClientImpl(statsStore, false, auth, addr, 1, 1*time.Millisecond, 1)
+	}
+
+	t.Run("no connection", func(t *testing.T) {
+		assert.Panics(t, func() {
+			// It's possible there is a redis server listening on 6379 in ci environment, so
+			// use a random port.
+			mkRedisClient("", "localhost:12345")
+		})
+	})
+
+	t.Run("ok", func(t *testing.T) {
+		redisSrv := mustNewRedisServer()
+		defer redisSrv.Close()
+
+		var client redis.Client
+		assert.NotPanics(t, func() {
+			client = mkRedisClient("", redisSrv.Addr())
+		})
+		assert.NotNil(t, client)
+	})
+
+	t.Run("auth fail", func(t *testing.T) {
+		redisSrv := mustNewRedisServer()
+		defer redisSrv.Close()
+
+		redisSrv.RequireAuth(redisAuth)
+
+		assert.Panics(t, func() {
+			mkRedisClient("", redisSrv.Addr())
+		})
+	})
+
+	t.Run("auth pass", func(t *testing.T) {
+		redisSrv := mustNewRedisServer()
+		defer redisSrv.Close()
+
+		redisSrv.RequireAuth(redisAuth)
+
+		assert.NotPanics(t, func() {
+			mkRedisClient(redisAuth, redisSrv.Addr())
+		})
+	})
```

Comment:
is it possible to test tls case as well?

Reply:
tls support is not available for now. Maybe we can do it in another pr when 
https://github.com/alicebob/miniredis/issues/155 solved ?

Reply:
okay, sound fair.